dashboard/app/controllers/lessons_controller.rb

```diff
@@ -41,14 +43,21 @@ def update
   private
 
   def lesson_params
+    # Convert camelCase params to snake_case. Right now this only works on
+    # top-level key names. This lets us do the transformation before calling
+    # .permit, so that we can use snake_case key names in our parameter list,
+    # because transform_keys returns a params object while deep_transform_keys
+    # returns a plain Hash.
+    lp = params.transform_keys(&:underscore)
+
     # for now, only allow editing of fields that cannot be edited on the
     # script edit page.
-    lp = params.permit(
+    lp = lp.permit(
       :overview,
```

Comment:
I would argue for having one list of params which is used both in `permit` here, as well as to populate the `lesson_data` object in `edit`. Case in point: why isn't `:name` in this list, isn't it editable?

Reply:
> I would argue for having one list of params which is used both in `permit` here, as well as to populate the `lesson_data` object in `edit`. 

I'm in favor of this. What do you think about this way of getting there, by using ActiveModel Serializers? https://github.com/code-dot-org/code-dot-org/pull/36894

> Case in point: why isn't `:name` in this list, isn't it editable?

`name` is actually not editable, due to the comment just above. it can only be edited on the script edit page in the script DSL.

Reply:
> > I would argue for having one list of params which is used both in `permit` here, as well as to populate the `lesson_data` object in `edit`.
> 
> I'm in favor of this. What do you think about this way of getting there, by using ActiveModel Serializers? #36894

tracking this as subtask of the current work in PLAT-280